src/jit/assertionprop.cpp

```diff
@@ -2093,7 +2093,7 @@ void Compiler::optAssertionGen(GenTreePtr tree)
         case GT_CALL:
             // A virtual call can create a non-null assertion. We transform some virtual calls into non-virtual calls
             // with a GTF_CALL_NULLCHECK flag set.
-            if ((tree->gtFlags & GTF_CALL_NULLCHECK) || ((tree->gtFlags & GTF_CALL_VIRT_KIND_MASK) != GTF_CALL_NONVIRT))
+            if ((tree->gtFlags & GTF_CALL_NULLCHECK) || tree->AsCall()->IsVirtual())
```

Comment:
It looks like you've replaced both:
```
(tree->gtFlags & GTF_CALL_VIRT_KIND_MASK) != GTF_CALL_NONVIRT)
```
and
```
call->gtFlags & (GTF_CALL_VIRT_VTABLE | GTF_CALL_VIRT_STUB)
```
with `IsVirtual()`, which is the former. Is there someplace that asserts that these are equivalent?

Reply:
Probably no assert, but `(GTF_CALL_VIRT_VTABLE | GTF_CALL_VIRT_STUB) == 0x30000000` so the expressions reduce to:
```
(tree->gtFlags & GTF_CALL_VIRT_KIND_MASK) != GTF_CALL_NONVIRT
=>
(tree->gtFlags & 0x30000000) != 0

call->gtFlags & (GTF_CALL_VIRT_VTABLE | GTF_CALL_VIRT_STUB)
=>
call->gtFlags & 0x30000000
```

Actually, it was this kind of thing that drove me crazy looking at these flags.